backend/services/reaction.py

```
@@ -0,0 +1,77 @@
+    if instance:
+        delta = 1 if action == "add" else -1
+        instance.like_count = max(0, instance.like_count + delta)
+        db.commit()
+
+
+def remove_like(existing_like, db: Session, like_data: LikeCreate):
+    db.delete(existing_like)
+    update_like_count(like_data, db, "remove")
+
+
+def add_like(like_data: LikeCreate, db: Session, current_user: User):
```

Ruff (0.8.2)

Missing return type annotation for public function `add_like`

(ANN201)